src/feditest/nodedrivers/imp/__init__.py

```diff
@@ -91,29 +85,21 @@ def perform_webfinger_query(
         if pair:
             ret_pair = HttpRequestResponsePair(first_request, current_request, pair.response)
             if ret_pair.response is not None:
-                if ret_pair.response.http_status == 200:
-                    if (
-                        not check_content_type
-                        or ret_pair.response.content_type() == "application/jrd+json"
-                        or ret_pair.response.content_type().startswith(
-                            "application/jrd+json;"
-                        )
-                    ):
-                        if ret_pair.response.payload is not None:
-                            json_string = ret_pair.response.payload.decode(
-                                encoding=ret_pair.response.payload_charset() or "utf8" )
-                            jrd = ClaimedJrd(json_string)
-                            if validate_jrd:
-                                try:
-                                    jrd.validate()
-                                except Exception as ex:
-                                    raise AssertionError(*ex.args[1:])
-                            return WebFingerQueryResponse(pair, jrd)
-                        raise WebFingerClient.WebfingerQueryFailedError(
-                            uri, ret_pair, "No payload"
-                        )
+                if ret_pair.response.http_status != 200:
+                    raise WebFingerClient.WebfingerQueryFailedError(uri, ret_pair, f"Invalid HTTP status: { ret_pair.response.http_status }")
+
+                if( ret_pair.response.content_type() != "application/jrd+json"
+                    and not ret_pair.response.content_type().startswith( "application/jrd+json;" )
+                ):
                     raise WebFingerClient.WebfingerQueryFailedError(uri, ret_pair, f"Invalid content type: { ret_pair.response.content_type() }")
-                raise WebFingerClient.WebfingerQueryFailedError(uri, ret_pair, f"Invalid HTTP status: { ret_pair.response.http_status }")
+
+                if ret_pair.response.payload is None:
+                    raise WebFingerClient.WebfingerQueryFailedError( uri, ret_pair, "No payload" )
+
+                json_string = ret_pair.response.payload.decode(encoding=ret_pair.response.payload_charset() or "utf8")
+                jrd = ClaimedJrd(json_string)
+                jrd.validate() # May throw JrdError
```

Comment:
This also looks like it will effectively change the default of `validate_jrd` to `True`, which previously caused issues. Again, I'll double-check that with some local testing.